Client/Frontend/Settings/SettingsViewController.swift

```diff
@@ -313,23 +313,48 @@ class SettingsViewController: TableViewController {
             BoolRow(
                 title: Strings.Private_Browsing_Only,
                 option: Preferences.Privacy.privateBrowsingOnly,
-                onValueChange: {
-                    Preferences.Privacy.privateBrowsingOnly.value = $0
-                    
-                    // Need to flush the table, hacky, but works consistenly and well
-                    let superView = self.tableView.superview
-                    self.tableView.removeFromSuperview()
-                    DispatchQueue.main.async {
-                        // Let shield toggle change propagate, otherwise theme may not be set properly
-                        superView?.addSubview(self.tableView)
-                        self.applyTheme(self.theme)
+                onValueChange: { value in
+
+                    if value {
+                        let alert = UIAlertController(title: Strings.Private_Browsing_Only, message: Strings.Private_Browsing_Only_Warning, preferredStyle: .alert)
+                        alert.addAction(UIAlertAction(title: Strings.CancelButtonTitle, style: .cancel, handler: { _ in
+                            DispatchQueue.main.async {
+                                self.toggleSwitch(on: false, section: self.privacySection, rowUUID: Preferences.Privacy.privateBrowsingOnly.key)
+                            }
+                        }))
+
+                        alert.addAction(UIAlertAction(title: Strings.OKString, style: .default, handler: { _ in
+                            Preferences.Privacy.privateBrowsingOnly.value = value
```

Comment:
This could should be moved to a block right above this. Then can just execute that block in either location, duplicating this much code is not ideal.